addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java

```diff
@@ -0,0 +1,345 @@
+import org.matmaul.freeboxos.login.LoginManager;
+import org.matmaul.freeboxos.phone.PhoneStatus;
+import org.matmaul.freeboxos.system.SystemConfiguration;
+import org.matmaul.freeboxos.wifi.WifiGlobalConfig;
+import org.openhab.binding.freebox.config.FreeboxServerConfiguration;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.freebox.FreeboxBindingConstants.*;
+
+/**
+ * The {@link FreeboxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * @author Gaël L'hopital - Initial contribution
+ */
+public class FreeboxHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(FreeboxHandler.class);
+    private ScheduledFuture<?> globalJob;
+    private ScheduledFuture<?> phoneJob;
+    
+    private static FreeboxOsClient fbClient = null;
+	private long uptime = -1;
+	
+	public FreeboxHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		try {
+			switch (channelUID.getId()) {
+				case LCDBRIGHTNESS : setBrightness(command);
+					break;
+				case LCDORIENTATION : setOrientation(command);
+					break;
+				case LCDFORCED : setForced(command);
+					break;
+				case WIFISTATUS : setWifiStatus(command);
+					break;
+				case REBOOT : setReboot(command);
+			}
+		} catch (FreeboxException e) {
+			logger.error(e.getMessage());
+		}
+	}
+	
+	@Override
+	public void initialize() {
+		FreeboxServerConfiguration configuration = getConfigAs(FreeboxServerConfiguration.class);
+		if (configuration.appToken != null && !configuration.appToken.isEmpty()) {
+			Bundle bundle = FrameworkUtil.getBundle(getClass());			
+			
+			fbClient = new FreeboxOsClient(
+						bundle.getSymbolicName(), /*  org.openhab.binding.freebox */
+						configuration.ipAddress);
+			
+			LoginManager loginManager = fbClient.getLoginManager();
+
+			loginManager.setAppToken(configuration.appToken);
+			try {
+				loginManager.openSession();
+				if (getThing().getStatus() == ThingStatus.OFFLINE) {
+		    		logger.debug("Setting status for thing '{}' to ONLINE", getThing().getUID());
+		    		getThing().setStatus(ThingStatus.ONLINE);
+		    	}
+				
+				if (globalJob == null || globalJob.isCancelled()) {
+					long polling_interval = Integer.parseInt(getConfigAs(FreeboxServerConfiguration.class).refreshInterval);
+					globalJob = scheduler.scheduleAtFixedRate(globalRunnable, 1, polling_interval, TimeUnit.SECONDS);
+				}
+				
+				if (phoneJob == null || phoneJob.isCancelled()) {
+					long polling_interval = Integer.parseInt(getConfigAs(FreeboxServerConfiguration.class).refreshPhoneInterval);
+					phoneJob = scheduler.scheduleAtFixedRate(phoneRunnable, 1, polling_interval, TimeUnit.SECONDS);
+				}
+			} catch (FreeboxException e) {
+				logger.error(e.getMessage());
```

Comment:
What if the login had failed here. Will it ever be retried (i.e. how to get from OFFLINE to ONLINE again)?